configuration/pih/liquibase/liquibase.xml

```diff
@@ -727,4 +727,86 @@
     </changeSet>
 
+     <changeSet id="20230309-fix-missing-medication-categories" author="dmdesimone">
+	    <preConditions onFail="MARK_RAN">
+			<not><sqlCheck expectedResult="0">
+					select count(*)  
+					from encounter e
+					inner join obs o on o.encounter_id = e.encounter_id and o.voided = 0 and o.concept_id = concept_from_mapping('PIH','9070') -- dispensing construct
+					where e.encounter_type =
+						(select encounter_type_id from encounter_type where uuid = 'cc1720c9-3e4c-4fa8-a7ec-40eeaad1958c')
+					and e.voided = 0
+					and not EXISTS
+						(select 1 from obs o2
+						where o2.obs_group_id = o.obs_id
+						and o2.voided = 0
+						and o2.concept_id = concept_from_mapping('PIH','1535'))
+					;
+			</sqlCheck></not>
+	    </preConditions>
+	    <comment> UHM-7028 fix data with medication categories missing </comment>
+		<sql>
+			DROP TEMPORARY TABLE IF EXISTS temp_missing_categories;
+			create temporary table temp_missing_categories
+			select 
+				e.encounter_id,
+				o.obs_datetime, 
+				o.obs_id "Dispensing_construct_obs_group_id", 
+				concept_name(om.value_coded,'en') "drug", 
+				o.person_id, 
+				o.location_id , 
+				o.creator , 
+				o.date_created  
+			from encounter e
+			inner join obs o on o.encounter_id = e.encounter_id and o.voided = 0 and o.concept_id = concept_from_mapping('PIH','9070') -- dispensing construct
+			inner join obs om on om.obs_group_id = o.obs_id and om.voided = 0 and om.concept_id = concept_from_mapping('PIH','1282') --  medication name
+			where e.encounter_type =
+				(select encounter_type_id from encounter_type where uuid = 'cc1720c9-3e4c-4fa8-a7ec-40eeaad1958c') -- hiv dispensing encounter
+			and e.voided = 0
+			and not EXISTS
+				(select 1 from obs o2
+				where o2.obs_group_id = o.obs_id
+				and o2.voided = 0
+				and o2.concept_id = concept_from_mapping('PIH','1535')) -- medication category
+			order by e.encounter_datetime desc
+			;
+			
+			insert into obs 
+				(person_id,
+				concept_id,
+				encounter_id,
+				obs_datetime,
+				location_id,
+				obs_group_id,
+				creator,
+				date_created,
+				voided,
+				uuid,
+				status,
+				value_coded)
+			select 
+				tmc.person_id,
+				concept_from_mapping('PIH','1535'), -- medication category
+				tmc.encounter_id,
+				tmc.obs_datetime,
+				tmc.location_id,
+				tmc.Dispensing_construct_obs_group_id,
+				tmc.creator,
+				tmc.date_created,
+				0,
+				uuid(),
+				'FINAL',
+				CASE tmc.drug 
+					when '3TC + AZT' then concept_from_mapping('PIH','3013') -- ARV #1
+					when 'ABC + 3TC' then concept_from_mapping('PIH','3013') -- ARV #1
+					when 'ATV/r'  then concept_from_mapping('PIH','2848') -- ARV #2
+					when 'Cotrimoxazole' then concept_from_mapping('PIH','3120') -- Prophylaxis drug
+					when 'DTG' then concept_from_mapping('PIH','2848') -- ARV #2
+					when 'Isoniazid' then concept_from_mapping('PIH','656') -- Isoniazid
+					when 'TDF + 3TC + DTG' then concept_from_mapping('PIH','3013') -- ARV #1
```

Comment:
I figure some of these are "best guess"?  Or are we skipping the ones that aren't 100% clear where they belong?

Reply:
These are all of them, but yes 2 of them are "best guess" but I will confirm with the program before committing.